polaris-service/src/main/java/org/apache/polaris/service/PolarisApplication.java

```diff
@@ -207,11 +208,15 @@ public void run(PolarisApplicationConfig configuration, Environment environment)
       csAware.setConfigurationStore(configurationStore);
     }
 
+    Boolean skipCredentialSubscopingIndirection =
+        configurationStore.getConfiguration(
+            null, PolarisConfiguration.SKIP_CREDENTIAL_SUBSCOPING_INDIRECTION.key);
     TaskHandlerConfiguration taskConfig = configuration.getTaskHandler();
     TaskExecutorImpl taskExecutor =
         new TaskExecutorImpl(taskConfig.executorService(), metaStoreManagerFactory);
     TaskFileIOSupplier fileIOSupplier =
-        new TaskFileIOSupplier(metaStoreManagerFactory, fileIOFactory);
+        new TaskFileIOSupplier(
+            metaStoreManagerFactory, fileIOFactory, skipCredentialSubscopingIndirection);
```

Comment:
I'd probably just pass the `PolarisConfigurationStore` directly to the `TaskFileIOSupplier` so that it can look up its own configuration keys.

Reply:
Thanks for your feedback